mm2src/coins/tx_history_storage/sql_tx_history_storage_v2.rs

```diff
@@ -0,0 +1,695 @@
+use crate::my_tx_history_v2::{GetHistoryResult, RemoveTxResult, TxHistoryStorage, TxHistoryStorageError};
+use crate::tx_history_storage::{token_id_from_tx_type, ConfirmationStatus, CreateTxHistoryStorageError,
+                                FilteringAddresses, GetTxHistoryFilters, WalletId};
+use crate::TransactionDetails;
+use async_trait::async_trait;
+use common::{async_blocking, PagingOptionsEnum};
+use db_common::sql_query::SqlQuery;
+use db_common::sqlite::rusqlite::types::Type;
+use db_common::sqlite::rusqlite::{Connection, Error as SqlError, Row, NO_PARAMS};
+use db_common::sqlite::{query_single_row, string_from_row, validate_table_name, CHECK_TABLE_EXISTS_SQL};
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use rpc::v1::types::Bytes as BytesJson;
+use serde_json::{self as json};
+use std::convert::TryInto;
+use std::sync::{Arc, Mutex};
+
+fn tx_history_table(wallet_id: &WalletId) -> String { wallet_id.to_sql_table_name() + "_tx_history" }
+
+fn tx_from_address_table(wallet_id: &WalletId) -> String { wallet_id.to_sql_table_name() + "_tx_from_address" }
```

Comment:
As per https://github.com/KomodoPlatform/atomicDEX-API/pull/1285#discussion_r892127284, address itself can be used as `key` for the entire history. I may still misunderstand something, but it's not really clear why this `from/to` separation is required at all.

Reply:
I at least assume that there can be one `JOIN` less.

Reply:
I thought about replacing `from_addresses_table` and `to_addresses_table` with the single `for_addresses_table`, but what if we need to query transactions that were sent **from** or **to** specific addresses **only**?
I don't know if it can be the case, so if you think that we can get by with just one `for_addresses_table`, then I'll do it.

Reply:
> but what if we need to query transactions that were sent from or to specific addresses only?

There is no such TX history use case for now. Also, if it's required, there can be `direction` flag signalling whether the transaction is incoming or outgoing. I guess it will be easier to handle with one `WHERE` instead of `JOIN`.

Reply:
Anyway, since we don't plan to query transactions by the filter like `from` or `to`, I'll replace these tables with a single one.

Reply:
I combined `tx_from_address` and `tx_to_address` tables into single `tx_address`